GitTfsTest/TestHelpers/AssertTimeout.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Sep.Git.Tfs.Test.TestHelpers
+{
+    public static class AssertTimeout
+    {
+        /// <summary>
+        /// Fails tests which would otherwise run indefinitely.
+        /// </summary>
+        public static void For(TimeSpan limit, Action actionToLimit, string message = null)
+        {
+            if (actionToLimit == null) throw new ArgumentNullException("actionToLimit");
+
+            var thread = new Thread(actionToLimit.Invoke)
+            {
+                Name = "AssertTimeout " + actionToLimit.Method.Name
+            };
+
+            thread.Start();
+
+            if (Debugger.IsAttached)
+            {
+                thread.Join();
```

Comment:
I don't remember what c# does with exceptions on a child thread. Will an exception raised by the child get thrown here, too? The [docs](https://msdn.microsoft.com/en-us/library/95hbf2ta.aspx) don't say, so my guess is that they don't get thrown by `Join` either. Should this re-throw errors from the child thread?